internal/input/changes.go

```diff
@@ -116,97 +110,205 @@ func closeChannels(accChannel, offChannel, trustChannel chan xdr.LedgerEntry) {
+		BatchStart: batchStart,
+		BatchEnd:   batchEnd,
+		Type:       xdr.LedgerEntryTypeAccount,
 	}
+	sendBatchToChannels(accBatch, accChannel, offChannel, trustChannel)
+
+	offBatch := ChangeBatch{
+		Changes:    offChanges.GetChanges(),
+		BatchStart: batchStart,
+		BatchEnd:   batchEnd,
+		Type:       xdr.LedgerEntryTypeOffer,
+	}
+	sendBatchToChannels(offBatch, accChannel, offChannel, trustChannel)
+
+	trustBatch := ChangeBatch{
+		Changes:    trustChanges.GetChanges(),
+		BatchStart: batchStart,
+		BatchEnd:   batchEnd,
+		Type:       xdr.LedgerEntryTypeTrustline,
+	}
+	sendBatchToChannels(trustBatch, accChannel, offChannel, trustChannel)
 }
 
 // StreamChanges runs a goroutine that reads in ledgers, processes the changes, and send the changes to the channel matching their type
-func StreamChanges(core *ledgerbackend.CaptiveStellarCore, start, end uint32, accChannel, offChannel, trustChannel chan xdr.LedgerEntry) {
+func StreamChanges(core *ledgerbackend.CaptiveStellarCore, start, end, batchSize uint32, accChannel, offChannel, trustChannel chan ChangeBatch, logger *log.Entry) {
 	if end != 0 {
-		for seq := start; seq <= end; seq++ {
-			sendLedgerChangesToChannels(seq, core, accChannel, offChannel, trustChannel)
+		totalBatches := uint32(math.Ceil(float64(end-start+1) / float64(batchSize)))
+		for currentBatch := uint32(0); currentBatch < totalBatches; currentBatch++ {
+			batchStart := start + currentBatch*batchSize
+			batchEnd := batchStart + batchSize
+			if batchEnd > end+1 {
+				batchEnd = end + 1
+			}
+
+			exportBatch(batchStart, batchEnd, core, accChannel, offChannel, trustChannel, logger)
 		}
 	} else {
-		currentLedger := start
+		batchStart := start
+		batchEnd := batchStart + batchSize
 		for {
-			sendLedgerChangesToChannels(currentLedger, core, accChannel, offChannel, trustChannel)
-			currentLedger++
+			exportBatch(batchStart, batchEnd, core, accChannel, offChannel, trustChannel, logger)
+			batchStart = batchEnd
+			batchEnd = batchStart + batchSize
 		}
 	}
 
 	closeChannels(accChannel, offChannel, trustChannel)
 }
 
 // ReceiveChanges reads in the ledger entries from the provided channels, transforms them, and adds them to the slice with the other transformed entries.
-func ReceiveChanges(accChannel, offChannel, trustChannel chan xdr.LedgerEntry, logger *log.Entry) ([]transform.AccountOutput, []transform.OfferOutput, []transform.TrustlineOutput) {
+func ReceiveChanges(accChannel, offChannel, trustChannel chan ChangeBatch, logger *log.Entry) ([]transform.AccountOutput, []transform.OfferOutput, []transform.TrustlineOutput) {
 	transformedAccounts := make([]transform.AccountOutput, 0)
 	transformedOffers := make([]transform.OfferOutput, 0)
 	transformedTrustlines := make([]transform.TrustlineOutput, 0)
-
+	accBatchRead, offBatchRead, trustBatchRead := false, false, false
 	for {
-
 		select {
-		case entry, ok := <-accChannel:
+		case batch, ok := <-accChannel:
 			if !ok {
 				accChannel = nil
```

Comment:
It might be useful to explain why we set the channel as `nil` because reading one entry failed.